src/v/transform/tests/transform_manager_test.cc

```diff
@@ -289,8 +289,30 @@ class TransformManagerTest : public ::testing::Test {
         _tracker = t.get();
         _manager = std::make_unique<manager<ss::manual_clock>>(
           std::move(r), std::move(t));
+        _manager->start().get();
+        // This allows us to wait for the seastar queue to be drained.
+        //
+        // This is used because we need to know when manual_clock tasks are
+        // enqueued to ensure that task execution is deterministic. Because in
+        // debug mode seastar randomizes task order, so there is no way to wait
+        // for those tasks to be executed outside of draining the seastar queue.
+        ss::set_idle_cpu_handler([this](ss::work_waiting_on_reactor) {
```

Comment:
I'm curious if you saw this technique in Redpanda source tree. I ask because we had to solve a similar problem at one point in the past and I don't recall that we had support from Seastar for that like you've managed to get with this code.

Reply:
I did not. I studied the reactor source in seastar a lot until I stumbled upon this. AFAIK this is would be the first usage of `ss::set_idle_cpu_handler` in Redpanda.

Reply:
So cool